site/docs/plugins/commands.md

```diff
@@ -5,10 +5,316 @@ next: false
+- RegEx support for command names
+
+All of these features are made possible because you will define one or more central command structures that define your bot's commands.
+
+## Basic Usage
+
+Before we dive in, take a look at how you can register and handle a command with the plugin:
+
+```js
+const myCommands = new Commands();
+
+myCommands.command("hello", "Say hello", (ctx) => ctx.reply(`Hello, world!`));
+
+bot.use(myCommands);
+```
+
+This registers a new "/start" command to your bot that will be handled by the given middleware.
+
+Now, let's get into some of the extra tools this plugin has to offer.
+
+## Importing
+
+First of all, we need to import the `Commands` class.
+
+::: code-group
+
+```ts [TypeScript]
+import { Commands, commands, type CommandsFlavor } from "@grammyjs/commands";
+```
+
+```js [JavaScript]
+const { Commands, commands } = require("@grammyjs/commands");
+```
+
+```ts [Deno]
+import {
+  Commands,
+  commands,
+  type CommandsFlavor,
+} from "https://deno.land/x/grammy_commands/mod.ts";
+```
+
+:::
+
+Now that that's settled, let's see how we can make our commands visible to our users.
+
+## Automatic Command Setting
+
+Once you defined your commands with an instance of the `Commands` class, you can call the `setCommands` method, which will register all the defined commands to your bot.
+
+```js
+const myCommands = new Commands()
+
+myCommands.command("hello", "say hello", (ctx) => ctx.reply("Hi there!"))
+myCommands.command("start", "Start the bot", (ctx) => ctx.reply("Starting..."))\
+
+bot.use(myCommands)
+
+await myCommands.setCommands(bot)
+```
+
+This will make it so every command you registered is displayed on the menu of a private chat with your bot, or whenever users type `/` on a chat your bot is a member of.
+
+## Scoped Commands
+
+Did you know you can allow different commands to be shown on different chats depending on the chat type, the language, and even the user status in a chat group?
+That's what Telegram call **Command Scopes**.
+
+The `Command` class returned by the `command` method exposes a method called `addToScope`.
+This method takes in a [BotCommandScope](/ref/types/botcommandscope) together with one or more handlers, and registers those handlers to be ran at that specific scope.
+
+You don't even need to worry about calling `filter`, the `addToScope` method will guarantee that your handler only gets called if the context is right.
+
+Here's an example of a scoped command:
+
+```js
+const myCommands = new Commands();
+
+myCommands
+  .command("start", "Initializes bot configuration")
+  .addToScope(
+    { type: "all_private_chats" },
+    (ctx) => ctx.reply(`Hello, ${ctx.chat.first_name}!`),
+  )
+  .addToScope(
+    { type: "all_group_chats" },
+    (ctx) => ctx.reply(`Hello, members of ${ctx.chat.title}!`),
+  );
+```
+
+The "start" command can now be called from both private and group chats, and it will give a different response depending on where it gets called from.
```

Comment:
```suggestion
The `start` command can now be called from both private and group chats, and it will give a different response depending on where it gets called from.
```